torch/csrc/utils/swap_thread_state.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+// RAII struct to change the current PyThreadState to a new PyThreadState
```

Comment:
A little explanation about what a PyThreadState here would be pretty useful. Similarly, another question I like answered in docs is "When would I want to use this?"